test/e2e/service.go

```diff
@@ -1145,12 +1145,93 @@ var _ = framework.KubeDescribe("Services", func() {
 		}
 		By("checking kube-proxy health check responses are correct")
 		for n, publicIP := range ips {
-			framework.Logf("Checking health check response for node %s, public IP %s", nodes.Items[n].Name, publicIP)
+			jig.Logf("Checking health check response for node %s, public IP %s", nodes.Items[n].Name, publicIP)
 			// HealthCheck should pass only on the node where num(endpoints) > 0
 			// All other nodes should fail the healthcheck on the service healthCheckNodePort
 			expectedSuccess := nodes.Items[n].Name == readyHostName
 			jig.TestHTTPHealthCheckNodePort(publicIP, healthCheckNodePort, "/healthz", expectedSuccess)
 		}
+
+		By("changing ESIPP annotation off and checking behaviour")
+		// Toggle ESIPP annotation off and check client ip and healthchecks
+		svc = jig.UpdateServiceOrFail(svc.Namespace, svc.Name, func(svc *api.Service) {
+			svc.ObjectMeta.Annotations[service.AnnotationExternalTraffic] =
+				service.AnnotationValueExternalTrafficGlobal
+		})
+		pollErr := wait.PollImmediate(framework.Poll, kubeProxyLagTimeout, func() (bool, error) {
+			By("reading clientIP using the TCP service's NodePort")
+			content := jig.GetHTTPContent(nodeIP, tcpNodePort, kubeProxyLagTimeout, "/clientip")
+			clientIP := content.String()
+			jig.Logf("ClientIP detected by target pod using NodePort is %s", clientIP)
+			By("reading clientIP using the TCP service's service port via its external VIP")
+			content = jig.GetHTTPContent(ingressIP, svcTcpPort, kubeProxyLagTimeout, "/clientip")
+			clientIP = content.String()
+			jig.Logf("ClientIP detected by target pod using VIP:SvcPort is %s", clientIP)
+			By("checking if Source IP is not preserved")
+			if strings.HasPrefix(clientIP, "10.") {
+				jig.Logf("Source IP was (correctly) not preserved when ESIPP annotation was off")
+				return true, nil
+			}
+			return false, fmt.Errorf("Source IP is the client IP")
+		})
+		if pollErr != nil {
+			jig.Failf("Source IP was preserved even when the ESIPP annotation was off")
+		}
+		By("checking health check node port annotation is cleared when ESIPP is toggled off")
+		if service.GetServiceHealthCheckNodePort(svc) > 0 {
+			jig.Failf("Service HealthCheck NodePort annotation still present")
+		}
+		By("checking kube-proxy health checks fail on all nodes when ESIPP annotation is off")
+		for n, publicIP := range ips {
+			jig.Logf("Checking health check response for node %s, public IP %s", nodes.Items[n].Name, publicIP)
+			// All nodes should fail the healthcheck on the service healthCheckNodePort
+			jig.TestHTTPHealthCheckNodePort(publicIP, healthCheckNodePort, "/healthz", false)
+		}
```

Comment:
it's worth checking, if we can, that the previously allocated nodeport is free by creating a test servie with that exact nodeport. Risk is that we need to ensure this test is run serially. It's already `[Slow]`, not sure if all Slow tests are also serial. 

Whatever you decide, please add a todo.